watch_advertisements/web_app.js

```diff
@@ -0,0 +1,141 @@
+/**
+ *  Copyright 2022 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// Espruino devices publish a UART service by default.
+const nordicUARTService = '6e400001-b5a3-f393-e0a9-e50e24dcca9e';
+const bodyTemperatureService = '00001809-0000-1000-8000-00805f9b34fb';
+const testCharacteristic = '0b301809-193e-11eb-adc1-0242ac120002';
+
+let gattServer = undefined;
+let device = undefined;
+let btDeviceId = null;
+let intervals = [];
+let previous_timstamp = null;
+const urlParam = 'bt_device_id';
+const btDeviceIdUrlParam = new URLSearchParams(window.location.search).get(urlParam);
+
+function disableButtons(disabled=true) {
+  $('btn_load_code').disabled = disabled;
+  $('btn_watch_advertisements').disabled = disabled;
+  $('btn_watch_advertisements_no_adv').disabled = disabled;
+}
+
+function onGattDisconnected(evt) {
+  logInfo(`Disconnected from GATT on device ${evt.target.name}.`);
+  assertFalse(gattServer.connected, 'Server connected');
+}
+
+/**
+ * Load the device code to the Espruino IDE.
+ */
+function loadEspruinoDeviceCode() {
+  fetch('device_code.js').then(response => response.text()).then(data => {
+    let url = 'http://www.espruino.com/webide?code=' + encodeURIComponent(data);
+    window.open(url, '_window');
+  });
+}
+
+function listenToAdvertisements(event) {
```

Comment:
Name this something like `checkAdvertisementEvent`?

Reply:
Done.